packages/components/src/components/single-select/shadow.tsx

```diff
@@ -0,0 +1,704 @@
+				const filterInput = document.getElementById('filter-input');
+				if (filterInput instanceof HTMLInputElement) {
+					filterInput.focus();
+					filterInput.setSelectionRange(0, 0);
+				}
+				this.setFilteredOptionsByQuery(this._inputValue);
+			}
+		}
+	};
+
+	private handleInputValue() {
+		if (Array.isArray(this.state._options) && this.state._options.length > 0 && !this.state._options.some((option) => option.label === this._inputValue)) {
+			this._inputValue = Array.isArray(this._options)
+				? (this.state._options.find((option) => (option as Option<string>).value === this.state._value)?.label as string)
+				: '';
+		}
+	}
+
+	private clearSelection() {
+		if (this.state._disabled) {
+			return;
+		} else {
+			this._focusedOptionIndex = -1;
+			this.state._value = '';
+			this._value = '';
+			this._inputValue = '';
+			this.state._hasValue = false;
+			this._filteredOptions = [...this.state._options];
+		}
+	}
+
+	private selectOption(option: Option<string>) {
+		this.state._value = option.value;
+		this._value = option.value;
+		this._inputValue = option.label as string;
+
+		this.state._hasValue = !!option;
+		this._filteredOptions = [...this.state._options];
+	}
+
+	private onInput(event: Event) {
+		const target = event.target as HTMLInputElement;
+		this._inputValue = target.value;
+		this._isOpen = true;
+		this.setFilteredOptionsByQuery(target.value);
+		this._focusedOptionIndex = -1;
+	}
+
+	private handleKeyDownDropdown(event: KeyboardEvent) {
+		if (event.key.length === 1 && /[a-z0-9]/i.test(event.key)) {
+			event.preventDefault();
+			this._isOpen = true;
+			this.focusSuggestionStartingWith(event.key);
+		}
+	}
+
+	private setFilteredOptionsByQuery(query: string) {
+		if (query?.trim() === '' || !query) {
+			this._filteredOptions = [...this.state._options];
+		} else if (Array.isArray(this.state._options) && this.state._options.length > 0 && query.length > 0) {
+			this._filteredOptions = this.state._options.filter((option) => {
+				return (option.label as string)?.toLowerCase()?.includes(query?.toLowerCase());
+			});
+		}
+		if (Array.isArray(this._filteredOptions)) {
+			this._showNoResultMessage = this._filteredOptions.length === 0;
+			if (this._filteredOptions.length === 0) {
+				const event = new CustomEvent('load-more-options');
+				this.host?.dispatchEvent(event);
+			}
+		}
+	}
+
+	private _focusedOptionIndex: number = -1;
+
+	private moveFocus(delta: number) {
+		if (!this._filteredOptions) {
+			return;
+		}
+		let newIndex = this._focusedOptionIndex + delta;
+
+		if (newIndex >= this._filteredOptions.length) {
+			newIndex = 0;
+		}
+
+		if (newIndex < 0) {
+			newIndex = this._filteredOptions.length - 1;
+		}
+
+		this._focusedOptionIndex = newIndex;
+		this.focusOption(this._focusedOptionIndex);
+	}
+
+	private focusOption(index: number) {
+		if (this.refOptions && this.refScrollContainer) {
+			this.refOptions.forEach((el) => {
+				if (el) el.classList.remove('highlighted');
```

Comment:
Könnten wir das direkt in JSX abbilden, statt über eine zusätzliche DOM-Manipulation?